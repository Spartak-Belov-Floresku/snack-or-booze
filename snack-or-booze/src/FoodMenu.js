import React from "react";
import { Link } from "react-router-dom";
import "./FoodMenu.css";
import {
  Card,
  CardBody,
  CardTitle,
  ListGroup,
  ListGroupItem
} from "reactstrap";


/*
  component will create snacks menu and displayed on the UI
*/
function FoodMenu({ snacks }) {
  return (
    <section className="col-md-4">
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            Food Menu
          </CardTitle>
          <ListGroup>
            {snacks.map(snack => (
              <Link to={`/snacks/${snack.id}`} key={snack.id}>
                <ListGroupItem>{snack.name}</ListGroupItem>
              </Link>
            ))}
          </ListGroup>
          <Link to="/"><ListGroupItem>Home page</ListGroupItem></Link>
        </CardBody>
      </Card>
    </section>
  );
}

export default FoodMenu;
