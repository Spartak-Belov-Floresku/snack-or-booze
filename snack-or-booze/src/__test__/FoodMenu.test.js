import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import SnackOrBoozeApi from "../Api";
import FoodMenu from '../FoodMenu';

let snacks;

/**befor ech tests get data from database */
beforeEach( async () => {
  function getSnacks() {
    return SnackOrBoozeApi.getSnacks();
  }   
  
  snacks = await getSnacks();

});


/** render FoodMenu component */
test('renders learn react link', () => {

    render(
        <BrowserRouter>
            <FoodMenu snacks={snacks} />
        </BrowserRouter>
    );
  
});

/** Matching snapshot of FoodMenu components */
it("matches snapshot Home component", () => {  
      
    const {asFragment} = render(
        <BrowserRouter>
            <FoodMenu snacks={snacks} />
        </BrowserRouter>
    );
    
    expect(asFragment()).toMatchSnapshot();
});

/** check if the menu exists on theFoodMenu page */
it("menu on the home page",  () => {

    const { queryByText } = render(
        <BrowserRouter>
            <FoodMenu snacks={snacks}/>
        </BrowserRouter>
    );
    expect(queryByText("Food Menu")).toBeInTheDocument();

});

/** checking if links to snack exists in FoodMenu bar*/
it("check link", () => {

    const { getByText } = render(
        <BrowserRouter>
            <FoodMenu snacks={snacks} />
        </BrowserRouter>
    );

    expect(screen.getByText('Nachos').closest('a')).toHaveAttribute('href', '/snacks/nachos');

});
