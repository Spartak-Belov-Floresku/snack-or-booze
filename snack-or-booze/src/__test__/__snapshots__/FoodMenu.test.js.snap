// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot Home component 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="font-weight-bold text-center card-title"
        >
          Food Menu
        </h5>
        <ul
          class="list-group"
        >
          <a
            href="/snacks/nachos"
          >
            <li
              class="list-group-item"
            >
              Nachos
            </li>
          </a>
          <a
            href="/snacks/hummus"
          >
            <li
              class="list-group-item"
            >
              Hummus
            </li>
          </a>
          <a
            href="/snacks/arugula-and-walnut-salad"
          >
            <li
              class="list-group-item"
            >
              Arugula and Walnut Salad
            </li>
          </a>
        </ul>
        <a
          href="/"
        >
          <li
            class="list-group-item"
          >
            Home page
          </li>
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
